src/utils/deferred.ts

```diff
@@ -19,8 +19,8 @@ export class Deferred<T> implements Promise<T> {
   #isFinished = false;
   #promise: Promise<T>;
 
-  #resolve: (value: T) => void = () => {};
-  #reject: (value: unknown) => void = () => {};
+  #resolve?: (value: T) => void;
+  #reject?: (value: unknown) => void;
```

Comment:
This are always present as they are set up in the constructor. The `?` should be able to be removed.

Reply:
```
🏃 [tsc] Running command "tsc --build src/tsconfig.json --pretty"
src/utils/deferred.ts:22:3 - error TS2564: Property '#resolve' has no initializer and is not definitely assigned in the constructor.

22   #resolve: (value: T) => void;
     ~~~~~~~~


Found 1 error.

❌ [tsc] Failed with exit status 1
```

Reply:
It's needed

Reply:
Yeah, I think it's because they are set up inside the `new Promise` :disappointed: